File: rules-cli/src/test/java/RuleEvaluatorTest.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.util.HashMap;
import java.util.Map;

class RuleEvaluatorTest {

    @Test
    void givenRuleWhenValidatingShouldReturnTrue() {
        Map<String, Object> variables = new HashMap<>();
        variables.put("x", 5);
        variables.put("y", 10);

        RuleEvaluator visitor = new RuleEvaluator("x + y = 15", variables, new HashMap<>());
        assertEquals("true", visitor.evaluate());
    }

    @Test
    void givenRuleWhenValidatingShouldReturnFalse() {
        Map<String, Object> variables = new HashMap<>();
        variables.put("x", 5);
        variables.put("y", 10);

        RuleEvaluator visitor = new RuleEvaluator("x + y = 16 return true", variables, new HashMap<>());
        assertEquals("false", visitor.evaluate());
    }

    @Test
    void givenExceptionRuleWhenValidatingShouldReturnErrorMessage() {
        Map<String, Object> variables = new HashMap<>();
        variables.put("x", 5);
        variables.put("y", 10);

        RuleEvaluator visitor = new RuleEvaluator("as0da9ad +", variables, new HashMap<>());
        assertTrue(visitor.evaluate().contains("error"));
    }
}

File: rules-cli/src/test/kotlin/RuleEvaluatorTest.kt
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class RuleEvaluatorTest {
    @Test
    fun `given a rule when validating it should return true`() {
        val visitor = RuleEvaluator("x + y = 15", mapOf("x" to 5, "y" to 10), mapOf())
        assertEquals("= true", visitor.evaluate())
    }
    @Test
    fun `given a rule when validating it should return false`() {
        val visitor = RuleEvaluator("x + y = 16 return true", mapOf("x" to 5, "y" to 10), mapOf())
        assertEquals("= false", visitor.evaluate())
    }
    @Test
    fun `given an exception rule when validating it should return error message`() {
        val visitor = RuleEvaluator("as0da9ad +", mapOf("x" to 5, "y" to 10), mapOf())
        assertEquals("error: Error at line 1:10 - mismatched input '<EOF>' expecting {STRING_NOT_SPECIAL_CHARS, '.', CURRENT_DATE, DATE_DIFF, 'abs', REGEX_STRIP, '(', K_NULL, DAY_OF_WEEK, ID, NUMERIC_LITERAL, BOOLEAN_LITERAL, SQUOTA_STRING}\n" +
                "as0da9ad +", visitor.evaluate())
    }
}